components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -3045,6 +2888,425 @@ subroutine check_values(Qv,T,i,timestepcount,force_abort,source_ind)
+   real(rtype), intent(in) :: xxlv 
+   real(rtype), intent(in) :: xlf
+   real(rtype), intent(in) :: dv 
+   real(rtype), intent(in) :: sc 
+   real(rtype), intent(in) :: mu 
+   real(rtype), intent(in) :: kap
+   real(rtype), intent(in) :: qv 
+   real(rtype), intent(in) :: qitot_incld 
+   real(rtype), intent(in) :: nitot_incld 
+
+   real(rtype), intent(out) :: qimlt 
+   real(rtype), intent(out) :: nimlt
+
+   real(rtype) :: qsat0
+
+   if (qitot_incld .ge.qsmall .and. t.gt.zerodegc) then
+      qsat0 = 0.622_rtype*e0/(pres-e0)
+
+      qimlt = ((f1pr05+f1pr14*sc**thrd*(rhofaci*rho/mu)**0.5_rtype)*((t-   &
+      zerodegc)*kap-rho*xxlv*dv*(qsat0-qv))*2._rtype*pi/xlf)*nitot_incld
+
+
+      qimlt = max(qimlt,0.)
+      nimlt = qimlt*(nitot_incld/qitot_incld)
+      
+   endif 
+
+
+end subroutine ice_melting
+
+
+subroutine wet_growth(rho,t,pres,rhofaci,    &
+f1pr05,f1pr14,xxlv,xlf,dv,kap,mu,sc,    &
+qv,qc_incld,qitot_incld,nitot_incld,qr_incld,    &
+           log_wetgrowth,qrcol,qccol,qwgrth,nrshdr,qcshd)
+
+   implicit none 
+
+   real(rtype), intent(in) :: rho 
+   real(rtype), intent(in) :: t 
+   real(rtype), intent(in) :: pres
+   real(rtype), intent(in) :: rhofaci 
+   real(rtype), intent(in) :: f1pr05
+   real(rtype), intent(in) :: f1pr14 
+   real(rtype), intent(in) :: xxlv
+   real(rtype), intent(in) :: xlf
+   real(rtype), intent(in) :: dv 
+   real(rtype), intent(in) :: kap 
+   real(rtype), intent(in) :: mu 
+   real(rtype), intent(in) :: sc
+   real(rtype), intent(in) :: qv 
+   real(rtype), intent(in) :: qc_incld 
+   real(rtype), intent(in) :: qitot_incld
+   real(rtype), intent(in) :: nitot_incld  
+   real(rtype), intent(in) :: qr_incld
+
+   logical, intent(inout) :: log_wetgrowth
+   real(rtype), intent(inout) :: qrcol 
+   real(rtype), intent(inout) :: qccol 
+   real(rtype), intent(inout) :: qwgrth
+   real(rtype), intent(inout) :: nrshdr 
+   real(rtype), intent(inout) :: qcshd 
+
+   real(rtype) :: qsat0, dum, dum1 
+
+
+   if (qitot_incld.ge.qsmall .and. qc_incld+qr_incld.ge.1.e-6_rtype .and. t.lt.zerodegc) then
+      qsat0  = 0.622_rtype*e0/(pres-e0)
+
+      qwgrth = ((f1pr05 + f1pr14*sc**thrd*(rhofaci*rho/mu)**0.5_rtype)*       &
+      2._rtype*pi*(rho*xxlv*dv*(qsat0-qv)-(t-zerodegc)*           &
+      kap)/(xlf+cpw*(t-zerodegc)))*nitot_incld
+
+      qwgrth = max(qwgrth,0._rtype)
+      dum    = max(0._rtype,(qccol+qrcol)-qwgrth)
+      if (dum.ge.1.e-10_rtype) then
+         nrshdr = nrshdr + dum*1.923e+6_rtype   ! 1/5.2e-7, 5.2e-7 is the mass of a 1 mm raindrop
+         if ((qccol+qrcol).ge.1.e-10_rtype) then
+            dum1  = 1._rtype/(qccol+qrcol)
+            qcshd = qcshd + dum*qccol*dum1
+            qccol = qccol - dum*qccol*dum1
+            qrcol = qrcol - dum*qrcol*dum1
+         endif
+         ! densify due to wet growth
+         log_wetgrowth = .true.
+      endif
+
+
+   end if 
+
+
+
+
+end subroutine wet_growth 
+
+
+subroutine ice_relaxation(rho,t,rhofaci,     &
```

Comment:
We should probably add "timescale" to the name here to make it clear this isn't for calculating a tendency directly.